cargo-ndk.sh

```diff
@@ -0,0 +1,285 @@
+#!/bin/sh
```

Comment:
Where does this file come from? It it related to [this repository with the same name](https://github.com/bbqsrc/cargo-ndk)? I would definitely prefer if we wouldn't have to vendor this script in our repository (and surely not in the root directory)

Reply:
this file is written by me and it is nothing to do with [cargo-ndk project](https://github.com/bbqsrc/cargo-ndk)

I wanna use [cargo-ndk project](https://github.com/bbqsrc/cargo-ndk), but it has bugs, finnaly I decide write my own version in shell. it is very simple and readable. 

I developed a tool call [ndk-pkg](https://github.com/leleliu008/ndk-pkg), if you want keep the workflow simple, you can use this tool.

```
name: Ubuntu

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  test:

    runs-on: ubuntu-20.04

    steps:
      - uses: actions/checkout@v2

      - run: curl -LO https://raw.githubusercontent.com/leleliu008/ndk-pkg/master/bin/ndk-pkg

      - run: chmod +x ndk-pkg

      - run: ./ndk-pkg update

      - name: create formula
        run:  |
          cat > ~/.ndk-pkg/repos.d/offical/formula/tealdeer.sh << EOF
          package set summary "Very fast implementation of tldr in Rust"
          package set git.url "https://github.com/dbrgn/tealdeer.git"
          package set src.url "dir://${{ github.workspace }}"
          package set license "Apache-2.0"
          package set dep.pkg "openssl"
          package set bsystem "cargo"
          EOF

      - run: ./ndk-pkg install tealdeer --tree
```

Reply:
I have port this project to `ndk-pkg`, here is a github action example https://github.com/leleliu008/ndk-pkg-formula-repository/runs/4992613697?check_suite_focus=true

Reply:
Ah that looks interesting :+1: If you have the tealdeer formula in your package base, we don't need to run `cargo-ndk` here, right? I am happy with applying the `android.patch` and committing this here, so you don't need to patch the code in the formula. Additionally, we can run a standard `cargo build` for the android target in CI so that it doesn't break out of the blue (testing is harder). Does that sound good to you?

Reply:
> If you have the tealdeer formula in your package base, we don't need to run cargo-ndk here, right?

In my ndk-pkg formula  repo, I use the stable tarball file https://github.com/dbrgn/tealdeer/archive/v1.5.0.tar.gz , not fetch from the github source tree HEAD

> we can run a standard `cargo build` for the android target in CI so that it doesn't break out of the blue (testing is harder).

just run `cargo build --target armv7-linux-androideabi` is not enough. some certain environment variables must be set, that's why I wrote `cargo-ndk.sh`.